cran/repodb.go

```diff
@@ -87,12 +92,15 @@ func (repoDb *RepoDb) Hash(rVersion string) string {
 }
 
 // GetPackagesFileURL provides the base URL for a package in a cran-like repo given the source type and version of R
-func GetPackagesFileURL(r RepoURL, st SourceType, rv RVersion) string {
+func GetPackagesFileURL(r *RepoDb, st SourceType, rv RVersion) string {
 	if st == Source {
-		return fmt.Sprintf("%s/src/contrib/PACKAGES", strings.TrimSuffix(r.URL, "/"))
+		return fmt.Sprintf("%s/src/contrib/PACKAGES", strings.TrimSuffix(r.Repo.URL, "/"))
 		// TODO: fix so isn't hard coded to 3.5 binaries
 	}
-	return fmt.Sprintf("%s/bin/%s/contrib/%s/PACKAGES", strings.TrimSuffix(r.URL, "/"), cranBinaryURL(rv), rv.ToString())
+	if r.RepoSuffix != "" {
+		return fmt.Sprintf("%s/bin/%s/PACKAGES", strings.TrimSuffix(r.Repo.URL, "/"), r.RepoSuffix)
```

Comment:
I don't think this is sufficient as a suffix - I see Suffix as truly being an additional component of the fully qualified "linux binary url" 

```
<uri>/bin/linux/<distro>/<reposuffix>/contrib/<rversion>/<pkg>_<version>_R_<platform>.tar.gz
```

In addition, if we do drop the linux/distro stuff, there still should be some differentiation around R versions to provide some consistency and make sure pkgr continues to operate in the multiple R version environment smoothly without having to manually set R Version numbers in the Suffix itself

For example, valid suffix examples might be:

```
20.03
metworx/20.06
```

```
/bin/linux/ubuntu/bionic/20.03/contrib/3.6
/bin/linux/ubuntu/bionic/20.03/contrib/4.0
/bin/linux/ubuntu/bionic/metworx/20.03/contrib/3.6
/bin/linux/ubuntu/bionic/metworx/20.06/contrib/4.0
```